src/contexts/membershipMintPage/MembershipMintPageContext.tsx

```diff
@@ -0,0 +1,113 @@
+import { Contract } from '@ethersproject/contracts';
+import { createContext, memo, ReactNode, useCallback, useContext, useMemo, useState } from 'react';
+
+function computeRemainingSupply(usedSupply: number, totalSupply: number) {
+  return Math.max(totalSupply - usedSupply, 0);
+}
+
+export type MembershipMintPageState = {
+  totalSupply: number;
+  remainingSupply: number;
+  price: number;
+};
+
+export const MembershipMintPageContext = createContext<MembershipMintPageState>({
+  totalSupply: 0,
+  remainingSupply: 0,
+  price: 0,
+});
+
+export const useMembershipMintPageState = (): MembershipMintPageState => {
+  const context = useContext(MembershipMintPageContext);
+  if (!context) {
+    throw new Error('Attempted to use MembershipMintPageContext without a provider');
+  }
+
+  return context;
+};
+
+type MembershipMintPageActions = {
+  setTotalSupply: (totalSupply: number) => void;
+  setRemainingSupply: (remainingSupply: number) => void;
+  getSupply: (contract: Contract, tokenId: number) => void;
+  getPrice: (contract: Contract, tokenId: number) => void;
+  setPrice: (price: number) => void;
+};
+
+const MembershipMintPageActionsContext = createContext<MembershipMintPageActions | undefined>(
+  undefined
+);
+
+export const useMembershipMintPageActions = (): MembershipMintPageActions => {
+  const context = useContext(MembershipMintPageActionsContext);
+  if (!context) {
+    throw new Error('Attempted to use MembershipMintPageActionsContext without a provider');
+  }
+
+  return context;
+};
+
+type Props = { children: ReactNode };
+
+const MembershipMintPageProvider = memo(({ children }: Props) => {
+  const [MembershipMintPageState, setMembershipMintPageState] = useState<MembershipMintPageState>({
+    totalSupply: 0,
+    remainingSupply: 0,
+    price: 0,
+  });
+
+  const setTotalSupply = useCallback((totalSupply: number) => {
```

Comment:
Supernit: if you have 3 separate setX functions, I'd prefer to keep the state separate (e.g. [totalSupply, setTotalSupply] etc) instead of one big object

Reply:
Then you can avoid having to do the object spreading / merging in each callback

Reply:
makes sense! done